src/Dev/Deprecation.php

```diff
@@ -234,14 +234,22 @@ public static function get_enabled()
 
     public static function isEnabled(): bool
     {
+        // Return early if disabled
         if (Environment::hasEnv('SS_DEPRECATION_ENABLED')) {
-            $envVar = Environment::getEnv('SS_DEPRECATION_ENABLED');
-            return self::varAsBoolean($envVar);
+            if (!Environment::getEnv('SS_DEPRECATION_ENABLED')) {
+                return false;
+            }
+        } elseif (!static::$currentlyEnabled) {
+            // Static property is ignored if SS_DEPRECATION_ENABLED was set
+            return false;
```

Comment:
```suggestion
        }
        
        if (!static::$currentlyEnabled) {
            return false;
```
Reads clearer as its own block, similar to how ` if (!Director::isDev()) {` is its own block

Reply:
We can't do this, because then the environment variable doesn't override the static variable - i.e. if `SS_DEPRECATION_ENABLED` is set to true, but `static::$currentlyEnabled` is set to false, we'll be returning false when we should be returning true.

Reply:
See https://github.com/silverstripe/silverstripe-framework/pull/11055#discussion_r1398611410 for an alternative which would work

Reply:
Superceded by https://github.com/silverstripe/silverstripe-framework/pull/11055#discussion_r1398611410